devito/interfaces.py

```diff
@@ -189,6 +197,7 @@ def __init__(self, *args, **kwargs):
             self.shape = (time_dim,) + self.shape
             # Store final instance in symbol cache
             self._cache_put(self)
+        MemmapManager.setup(self, *args, **kwargs)
```

Comment:
Yes, effectively we are copying the entire state (`__dict__`) from the existing object by reference, so we are effectively creating a new symbol but attached to the cached object. This means if we have a cache hit, we inherit the already existing pointers, memmap settings, etc, while in the cache miss case we go through the constructor hierarchy and `DenseData` already calls the setup.